frame/system/benchmarking/src/lib.rs

```diff
@@ -0,0 +1,180 @@
+
+	set_heap_pages {
+		// Heap page size
+		let i in 0 .. u32::max_value();
+	}: _(RawOrigin::Root, i.into())
+
+	// `set_code` was not benchmarked because it is pretty hard to come up with a real
+	// Wasm runtime to test the upgrade with. But this is okay because we will make
+	// `set_code` take a full block anyway.
+
+	set_code_without_checks {
+		// Version number
+		let b in 0 .. 16_384;
+		let code = vec![1; b as usize];
+	}: _(RawOrigin::Root, code)
+	verify {
+		let current_code = storage::unhashed::get_raw(well_known_keys::CODE).ok_or("Code not stored.")?;
+		assert_eq!(current_code.len(), b as usize);
+	}
+
+	set_changes_trie_config {
+		let d in 0 .. 1000;
+
+		let digest_item = DigestItemOf::<T>::Other(vec![]);
+
+		for i in 0 .. d {
+			System::<T>::deposit_log(digest_item.clone());
+		}
+		let changes_trie_config = ChangesTrieConfiguration {
+			digest_interval: d,
+			digest_levels: d,
+		};
+	}: _(RawOrigin::Root, Some(changes_trie_config))
+	verify {
+		assert_eq!(System::<T>::digest().logs.len(), (d + 1) as usize)
+	}
+
+	set_storage {
+		let i in 1 .. 1000;
+
+		// Set up i items to add
+		let mut items = Vec::new();
+		for j in 0 .. i {
+			let hash = (i, j).using_encoded(T::Hashing::hash).as_ref().to_vec();
+			items.push((hash.clone(), hash.clone()));
+		}
+	}: _(RawOrigin::Root, items)
+	verify {
+		let last_hash = (i, i - 1).using_encoded(T::Hashing::hash);
+		let value = storage::unhashed::get_raw(last_hash.as_ref()).ok_or("No value stored")?;
+		assert_eq!(value, last_hash.as_ref().to_vec());
+	}
+
+	kill_storage {
+		let i in 1 .. 1000;
+
+		// Add i items to storage
+		let mut items = Vec::new();
+		for j in 0 .. i {
+			let hash = (i, j).using_encoded(T::Hashing::hash).as_ref().to_vec();
+			storage::unhashed::put_raw(&hash, &hash);
+			items.push(hash);
+		}
+
+		// We will verify this value is removed
+		let last_hash = (i, i - 1).using_encoded(T::Hashing::hash);
+		let value = storage::unhashed::get_raw(last_hash.as_ref()).ok_or("No value stored")?;
+		assert_eq!(value, last_hash.as_ref().to_vec());
+
+	}: _(RawOrigin::Root, items)
+	verify {
+		assert_eq!(storage::unhashed::get_raw(last_hash.as_ref()), None);
+	}
+
+	kill_prefix {
+		let p in 1 .. 1000;
+
+		let prefix = p.using_encoded(T::Hashing::hash).as_ref().to_vec();
+		// add i items that share a prefix
+		for i in 0 .. p {
+			let hash = (p, i).using_encoded(T::Hashing::hash).as_ref().to_vec();
+			let key = [&prefix[..], &hash[..]].concat();
+			storage::unhashed::put_raw(&key, &key);
+		}
+
+		// We will verify this value is removed
+		let last_hash = (p, p - 1).using_encoded(T::Hashing::hash).as_ref().to_vec();
+		let last_key = [&prefix[..], &last_hash[..]].concat();
+		let value = storage::unhashed::get_raw(&last_key).ok_or("No value stored")?;
+		assert_eq!(value, last_key);
+
+	}: _(RawOrigin::Root, prefix)
+	verify {
+		assert_eq!(storage::unhashed::get_raw(&last_key), None);
+	}
+
+	suicide {
+		let n in 0 .. 1000;
```

Comment:
You might want to set `n` to values greater than 0 because of the check in `verify`?